PreSignUpLambdaHook/index.mjs

```diff
@@ -1,35 +1,26 @@
-﻿import pg from "pg";
-import fs from "fs";
+﻿import pg from 'pg';
+import fs from 'fs/promises';
 
-export const handler = async (event, context, callback) => {
+const checkEmailAddress = async (emailAddress) => {
     const client = new pg.Client({
         ssl: {
-            rejectUnauthorized: false,
-            ca: fs.readFileSync("certs/eu-west-2-bundle.pem").toString(),
+            ca: await fs.readFile('certs/eu-west-2-bundle.pem', 'utf-8'),
         },
     });
-    await client.connect();
-    const res = await client.query("SELECT email_address FROM local_authorities;");
 
-    const allowedEmailAddresses = res.rows.map(row => row.email_address);
+    await client.connect();
+    const res = await client.query(
+        'SELECT "EmailAddress" FROM "Users" WHERE LOWER("EmailAddress")=LOWER($1)',
+        [emailAddress],
+    );
+    await client.end();
 
-    // case-insensitive string comparison, see
-    // https://stackoverflow.com/questions/2140627/how-to-do-case-insensitive-string-comparison
-    if (!allowedEmailAddresses.some(
-        ea => {
-            const comparison = ea.localeCompare(
-                event.request.userAttributes.email,
-                "en-GB",
-                { sensitivity: "accent" }
-            );
-            return comparison === 0;
-        })
-    ) {
-        var error = new Error("Email address not allowed");
-        callback(error, event);
+    if (res.rowCount === 0) {
+        throw new Error('Email address not found');
     }
+};
 
-    callback(null, event);
-
-    await client.end();
+export const handler = async (event) => {
+    await checkEmailAddress(event.request.userAttributes.email);
+    return event;
```

Comment:
If you use `async`/`await` then you don't need to mess around with the callback

Reply:
Does it still handle the error correctly? I assumed that calling the callback with the error argument was doing something clever

Reply:
It's the old callback style of writing async code - it's equivalent, you just have to choose one or the other.

See https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html